---
title: "Australian Eating Survey for ACRC and QTAB microbiota study: EDA and PCs"
author: "by Chloe Yap - `r format(Sys.time(), '%d %B %Y')`"
output:
  epuRate::PCTG:
    toc: TRUE
    code_folding: "hide"
---

<br><br><br><br>

>Autism CRC Australian Eating Survey Data (ACRC + QTAB): Exploratory data analysis and PCs

# Loading packages

```{r, message=FALSE}

library(rmarkdown)    # install.packages("rmarkdown") 
library(data.table)
library(epuRate)      # devtools::install_github("holtzy/epuRate", force=TRUE)
library(DT)
library(ggplot2)
library(plotly)
library(tidyverse)
library(magrittr) # mutate_at()
library(purrr) # map()
library(corrplot)
library(caret)
library(naniar) # replace_with_na_all()
library(seqinr) # swap()
library(reshape2) # melt()
library(ggfortify) # pca, but for this need quantitative data?
library(polycor) # for polychoric correlation
library(ggbiplot) # for pca
library(factoextra) # for pca
library(compositions) # for clr
library(car) # for 
library(vegan) # for

```


# Loading data
***
Loading ACRC AES data and installing packages

```{r, message=FALSE, include = FALSE}
# Directories

## ACRC
acrc_dir <- "/Users/uqcyap3/Documents/Research/ASD/Data/Phenotype/AustralianAutismBiob-AccessRequestJakeGra_DATA_2019-03-15_2127_update191125_fillNA.rds"
acrc_pheno_dir <- "/Users/uqcyap3/Documents/Research/ASD/Data/Phenotype/extracted_pheno/cognitive_crossancestry_child.pheno"

## QTAB
# qtab_dir <- "/Users/uqcyap3/Documents/Research/ASD/Data/Diet/QTAB_AES/Wray 2019_09_17.csv"
qtab_dir <- "/Users/uqcyap3/Documents/Research/ASD/Data/Diet/QTAB_AES/Wray_2019_09_17_valuename.csv"
qtab_key_dir <- "/Users/uqcyap3/Documents/Research/ASD/Data/Diet/QTAB_AES/QTB_Stool_SentToMicroba_2019_DATA ENTRY.csv"
qtab_sex_dir <- "/Users/uqcyap3/Documents/Research/ASD/Data/QTAB/QTAB_Sample_Demog_20181205.csv"

## Key/ID data (Microba manifest)
ids_dir <- "/Users/uqcyap3/Documents/Research/ASD/Data/Stool/1_selection/20190306_ACRC_QTAB_Microba_Manifest.csv"
## Individuals used in analysis
ids_analysis_dir <- "/Users/uqcyap3/Documents/Research/ASD/Data/Stool/1_selection/181212_ACRC_QTAB_stool_demographics_split.tab"

## IDs for output
ids_fid_dir <- "/Users/uqcyap3/Documents/Research/ASD/Data/Stool/2_microba_output/AutismCRC metadata.csv"
ids_microba_dir <- "/Users/uqcyap3/Documents/Research/ASD/Data/Stool/ACRC_QTAB_Microba_Sample_IDs_long.csv"

# Read in data
# pheno <- readRDS(acrc_dir)
source("/Users/uqcyap3/Documents/GitHub/ASD/ACRC/pheno/phenofile_updates.R")
acrc <- pheno
qtab <- read.csv(qtab_dir, header = T, as.is = T)
qtab_key <- read.csv(qtab_key_dir, header = T, as.is = T)
qtab_sex_dob <- read.csv(qtab_sex_dir, header = T, as.is = T)
ids <- read.csv(ids_dir, header = T, as.is = T)
ids_analysis <- read.delim(ids_analysis_dir, header = T, as.is = T)
ids_fid <- read.csv(ids_fid_dir, header = T, as.is = T)
ids_microba <- read.csv(ids_microba_dir, header = T, as.is = T)

# Bristol Stool Chart data
qtab_bristol <- read.csv("/Users/uqcyap3/Documents/Research/ASD/Data/QTAB/3_metagenomics/QTB_Stool_SentToMicroba_2019_Chloe_bristolstoolchart.csv", header = T, as.is = T)
qtab_bristol <- qtab_bristol %>% dplyr::select("IID", "bristol_stool_chart_qtab") %>% filter(!is.na(bristol_stool_chart_qtab))
pheno_aab <- read.delim(acrc_pheno_dir)
aab_bristol <- pheno_aab %>% dplyr::select("IID", "bristol_stool_chart") %>% filter(!is.na(bristol_stool_chart))
colnames(aab_bristol) <- c("IID", "bristol_stool_chart_aab")

```

# Clean data

## Prune down data file to relevant columns and individuals

### ACRC

```{r, message=FALSE, include = FALSE}

# Take relevant columns
demo_col <- c("id", "participant_type", "participant_gender")
acrc_col <- colnames(acrc)
acrc_col <- acrc_col[grep("^es_", acrc_col)]

# Extract these columns
acrc <- acrc %>% dplyr::select(c(demo_col, acrc_col))
#acrc <- acrc %>% filter(!is.na(es_survey_id)) # use this is you want to include more individuals in the analysis. Would also need to do the equivalent for QTAB
acrc <- acrc %>% filter(id %in% ids$ID) # ?hold off on getting the IDs of metagenomics subset --> increase power
acrc <- acrc %>% dplyr::select(-es_age)

# Rename columns
colnames(acrc) <- gsub("^es_", "", colnames(acrc))
colnames(acrc)[which(colnames(acrc) == "participant_gender")] <- "sex"

# ACRC errors:
# Errors - have gone through the pheno file and manually altered these
# DOB errors (manually entered from the es_comments field)
acrc <- acrc %>% mutate(dob=replace(dob, (id == 3305120), "5/3/15"))
acrc <- acrc %>% mutate(dob=replace(dob, (id == 4436422), "30/5/10"))
acrc <- acrc %>% mutate(dob=replace(dob, (id == 1193664), "25/8/10"))
acrc <- acrc %>% mutate(dob=replace(dob, (id == 3305153), "30/11/06"))
acrc <- acrc %>% mutate(dob=replace(dob, (id == 1101709), "26/6/14"))
acrc <- acrc %>% mutate(dob=replace(dob, (id == 4406484), "25/8/11"))
acrc <- acrc %>% mutate(dob=replace(dob, (id == 1131625), "23/4/14"))
acrc <- acrc %>% mutate(dob=replace(dob, (id == 1193640), "22/8/14"))
acrc <- acrc %>% mutate(dob=replace(dob, (id == 1193667), "22/8/14"))
acrc <- acrc %>% mutate(dob=replace(dob, (id == 1193656), "19/1/11"))
acrc <- acrc %>% mutate(dob=replace(dob, (id == 3392390), "18/11/13"))
acrc <- acrc %>% mutate(dob=replace(dob, (id == 1101675), "16/9/01"))
acrc <- acrc %>% mutate(dob=replace(dob, (id == 1131685), "15/8/11"))
acrc <- acrc %>% mutate(dob=replace(dob, (id == 1131480), "8/7/10"))
acrc <- acrc %>% mutate(dob=replace(dob, (id == 3305120), "5/3/15"))

# Sex from genotyping (see phenotype_updates.R)
acrc[which(acrc$id == 1101238),"sex"] <- 1
acrc[which(acrc$id == 1101481),"sex"] <- 2
acrc[which(acrc$id == 1101447),"sex"] <- 1
acrc[which(acrc$id == 1101494),"sex"] <- 1
acrc[which(acrc$id == 3312757),"sex"] <- 2
acrc[which(acrc$id == 2227220),"sex"] <- 1
acrc[which(acrc$id == 1123378),"sex"] <- 1
acrc[which(acrc$id == 2215013),"sex"] <- 2
acrc[which(acrc$id == 4424977),"sex"] <- 1
acrc[which(acrc$id == 1101480),"sex"] <- 1
acrc[which(acrc$id == 2215039),"sex"] <- 2
acrc[which(acrc$id == 3312689),"sex"] <- 2
acrc[which(acrc$id == 2215055),"sex"] <- 2
acrc[which(acrc$id == 3312618),"sex"] <- 2
acrc[which(acrc$id == 3312741),"sex"] <- 2
acrc[which(acrc$id == 3312746),"sex"] <- 2
acrc[which(acrc$id == 3312697),"sex"] <- 2
acrc[which(acrc$id == 3312724),"sex"] <- 2
acrc[which(acrc$id == 3312750),"sex"] <- 2
acrc[which(acrc$id == 2215002),"sex"] <- 2
acrc[which(acrc$id == 2232570),"sex"] <- 2
acrc[which(acrc$id == 1116495),"sex"] <- 2
acrc[which(acrc$id == 1123661),"sex"] <- 1
acrc[which(acrc$id == 3305088),"sex"] <- 2

# BMI
acrc <- acrc %>% mutate(weight=replace(weight, (id == 1193640), "16"))
acrc <- acrc %>% mutate(height=replace(height, (id == 1193640), "98.9"))
acrc <- acrc %>% mutate(weight=replace(weight, (id == 1193660), "21.3"))
acrc <- acrc %>% mutate(height=replace(height, (id == 1193660), "120"))
acrc <- acrc %>% mutate(weight=replace(weight, (id == 1131696), "58.2"))
acrc <- acrc %>% mutate(height=replace(height, (id == 1131696), "159"))

```


### QTAB

```{r, message=FALSE, include = FALSE}
# AES file changes
# 1. Before joining, correct any errors
# STUDY_ID 258230, 258041 (CTG_COLLID 603109, 603110 / RPN ?) - AES not returned
# STUDY_ID 258437, 258992 (CTG_COLLID 602952, 603427 / RPN 105078, 105387) - missing AES for some reason 
# a) 258437 was due to mistyped RN --> change to 105078
# b) 258992/603427/105387 DOB 13/11/2006 in qtab_key. 
# - In QTAB_Sample_Demog_20181205.xlsx file, 2 IDs with 13/11/2006 DOB - 258320/603428 and 258992/603427. 
# - In qtab (AES), 2 IDs with 2006-11-13 DOB - 105389 and 105390
# - In qtab_key, 1 ID with RPN 105390 (258320/603428) 0 IDs with RPN 105389
# - Therefore, deduce that RPN 105387 in the qtab AES file is mistyped
# - --> change qtab 105389 to 105387

# 2. Change AES participant_ID (6-digit, starting with 10/12*****) to STUDY_ID (6-digit, starting with 25****)
# 3. Join anthropometrics
# 4. Join sex

qtab[which(qtab$participant_id == 258437),"participant_id"] <- 105078
qtab[which(qtab$participant_id == 105389),"participant_id"] <- 105387

qtab_demo <- qtab_key %>% dplyr::select(c("RPN", "STUDY_ID", "HEIGHT.cms.", "WEIGHT.kg."))
colnames(qtab_demo) <- c("participant_id", "id", "height", "weight")

qtab <- qtab %>% dplyr::select(-c("survey_id", "weight", "height"))
qtab <- join(qtab_demo, qtab)

qtab_sex_dob <- qtab_sex_dob %>% dplyr::select("ID", "sex", "dob")
colnames(qtab_sex_dob) <- c("id", "sex", "dob")
qtab <- qtab %>% dplyr::select(-c("dob"))
qtab <- join(qtab_sex_dob, qtab)
qtab$dob <- gsub("/20", "/", qtab$dob)
qtab$sex[which(qtab$sex == "M")] <- 1
qtab$sex[which(qtab$sex == "F")] <- 2

```

### Combine dataframes

```{r, message=FALSE}

acrc <- acrc %>% dplyr::select(-c("consent", "comments", "survey_id"))

qtab <- qtab %>% dplyr::select(-c("consent"))
qtab$participant_type <- "UNR_QTAB"

# Join dataframes (just use this function to get the columns in the right order essentially)
aes <- join(acrc, qtab, type = "full")

# Filter down to individuals used in analysis (ie. not all QTAB)
aes <- aes %>% filter(id %in% ids_analysis$Participant.ID)

```

## Checks and cleans

- N.B. QTAB date_started/date_finished incomplete. The ones that I do have are generally finished in July 2019 (ie. 08/07/2019). So have filled aes$date_finished with this.

```{r, message=FALSE}

# Rename participant_type
aes <- aes %>% mutate(participant_type=replace(participant_type, (participant_type %in% 1:2), "ASD"))
aes <- aes %>% mutate(participant_type=replace(participant_type, (participant_type == 3), "SIB"))
aes <- aes %>% mutate(participant_type=replace(participant_type, (participant_type == 4), "UNR_ACRC"))

# Create participant_group
aes$participant_group <- "NA"
aes <- aes %>% mutate(participant_group=replace(participant_group, (participant_type == "ASD"), "ASD"))
aes <- aes %>% mutate(participant_group=replace(participant_group, (participant_type == "SIB"), "SIB"))
aes <- aes %>% mutate(participant_group=replace(participant_group, (participant_type %in% c("UNR_ACRC", "UNR_QTAB")), "UNR"))

# Rename sex
# aes <- aes %>% mutate(sex=replace(sex, (sex == 1), "M"))
# aes <- aes %>% mutate(sex=replace(sex, (sex == 2), "F"))

# Check DOB
# Convert to standard format
aes$dob <- as.Date(aes$dob, format="%d/%m/%y")
aes$date_finished[which(aes$participant_type == "UNR_QTAB")] <- "8/7/19"
aes$date_finished <- as.Date(aes$date_finished, format="%d/%m/%y")
aes$age <- round(as.numeric(difftime(aes$date_finished, aes$dob, units = "days"))/365,2)
plot(aes$age)
min(aes$age[which(!is.na(aes$age))])
max(aes$age[which(!is.na(aes$age))])

# Check BMI is within limits
aes$height <- as.numeric(aes$height)
aes$weight <- as.numeric(aes$weight)
aes$bmi <- aes$weight/(aes$height/100)^2
plot(aes$bmi[which(!is.na(aes$bmi))])
min(aes$bmi[which(!is.na(aes$bmi))])
max(aes$bmi[which(!is.na(aes$bmi))])

```

```{r, message=FALSE, include = FALSE}

# Remove missing dietary data
## 1193652 doesn't have all data --> did analyses to restrict to what 1193652 has available
aes_1193652 <- aes %>% filter(!id %in% c(4494903, 258230))
## ACRC: 4494903,1193652 / QTAB: 258230
aes <- aes %>% filter(!id %in% c(4494903,1193652,258230))

# Remove 4494902 (Smith Magenis syndrome incorrectly assigned to UNR group)
aes_1193652 <- aes_1193652 %>% filter(!id %in% c(4494902))
aes <- aes %>% filter(!id %in% c(4494902))

# Add month
aes_1193652$date_finished <- as.Date(aes_1193652$date_finished, "%d/%m/%y")
aes_1193652$month <- format(aes_1193652$date_finished, "%m")

# Format so that the output is a neater file in .csv form
aes$q_food_other <- gsub(",", "_", aes$q_food_other)
aes$q_food_other <- gsub(" ", "_", aes$q_food_other)
aes$q_food_other <- gsub("\n", "_", aes$q_food_other)

```

## Reorder levels of food frequency columns

- exclude "milk" and "bread_type": not as frequency variables (if you want to include, need to add back in to the below)

When running order_food_levels(), the following variables fail the check:
- "milk": Not in frequency levels --> exclude
- "bread_type": Not in frequency levels --> exclude
- "grapes_etc": 
		[1] "1 - 3 per month"       "1 per week"            "2 - 4 per week"
		[4] "31"                    "5 or more per week"    "Less than 1 per month"
    - There's a stray "31". Check other fields with similar levels (eg. mayonnaise, nuts)
    - unique(aes$nuts). Missing field is "Never"

### Function

```{r}

order_food_levels <- function(x) {

# Order the factor levels
# Rules:
# - Never = 0
# - month / week / day
	# - less
	# - numbers in order
	# - more

	tmp <- as.factor(x)
	tmp.levels <- levels(tmp) # Tests: order_food_levels(acrc$breakfast_cereal)
	
	never <- grep("Never", tmp.levels)
	month <- grep("month", tmp.levels)
	week <- grep("week", tmp.levels)
	day <- grep("day", tmp.levels)
	
	output = NULL
	
	if (length(never) == 1) {
		output <- tmp.levels[never]
	}
	
	if (length(month > 1)) {
		# Take month variables
		tmp.levels_month <- tmp.levels[month]

		# Take Less/More: immediately put at the start/end
		# Take levels within each time period bracket
		less <- grep("Less", tmp.levels_month)
		more <- grep("More", tmp.levels_month)
		if (length(less) > 0 | length(more > 0)) {
			tmp.levels_month2 <- tmp.levels_month[-c(less, more)]			
		} else {
			tmp.levels_month2 <- tmp.levels_month
		}
		# Take numeric factors and extract the numbers
		num <- as.numeric(gsub("([0-9]+).*$", "\\1", tmp.levels_month2))

		# Order the indices within the time category
		# Put the NA first (eg. "Once per week" is less frequent/goes before other numbers)
		tmp.levels_month2_ord <- tmp.levels_month2[order(num, na.last = FALSE)]
		monthlev <- c(tmp.levels_month[less], tmp.levels_month2_ord, tmp.levels_month[more])
		output <- c(output, monthlev)
	}	
	
	if (length(week > 1)) {
		# Take week variables
		tmp.levels_week <- tmp.levels[week]

		# Take Less/More: immediately put at the start/end
		# Take levels within each time period bracket
		less <- grep("Less", tmp.levels_week)
		more <- grep("More", tmp.levels_week)
		if (length(less) > 0 | length(more > 0)) {
			tmp.levels_week2 <- tmp.levels_week[-c(less, more)]			
		} else {
			tmp.levels_week2 <- tmp.levels_week
		}

		# Take numeric factors and extract the numbers
		num <- as.numeric(gsub("([0-9]+).*$", "\\1", tmp.levels_week2))

		# Order the indices within the time category
		# Put the NA first (eg. "Once per week" is less frequent/goes before other numbers)
		tmp.levels_week2_ord <- tmp.levels_week2[order(num, na.last = FALSE)]
		weeklev <- c(tmp.levels_week[less], tmp.levels_week2_ord, tmp.levels_week[more])
		output <- c(output, weeklev)
	}

	if (length(day > 1)) {
		# Take day variables
		tmp.levels_day <- tmp.levels[day]

		# Take Less/More: immediately put at the start/end
		# Take levels within each time period bracket
		less <- grep("Less", tmp.levels_day)
		more <- grep("More", tmp.levels_day)
		if (length(less) > 0 | length(more > 0)) {
			tmp.levels_day2 <- tmp.levels_day[-c(less, more)]			
		} else {
			tmp.levels_day2 <- tmp.levels_day
		}

		# Take numeric factors and extract the numbers
		num <- as.numeric(gsub("([0-9]+).*$", "\\1", tmp.levels_day2))

		# Order the indices within the time category
		# Put the NA first (eg. "Once per week" is less frequent/goes before other numbers)
		tmp.levels_day2_ord <- tmp.levels_day2[order(num, na.last = FALSE)]
		daylev <- c(tmp.levels_day[less], tmp.levels_day2_ord, tmp.levels_day[more])
		output <- c(output, daylev)
	}
	# Checks
	# - to capture all levels
	# - to show all unique
	print(c(length(output) == length(tmp.levels)) & length(unique(output)) == length(tmp.levels))
	
	return(output)

}

```

### Re-format

```{r, message=FALSE}

food_col <- c("snacks","dairy","daily_softdrinks","diet_softdrink","softdrink","water_bottled_etc","juice","cordials","tea_coffee","beer","wine","spirits",
	"flavoured_milk","plain_milk","cream","icecream","frozen_yoghurt","yoghurt","cottage_cheese","cheese","cheese_spread",
	"muesli","porridge","breakfast_cereal","bread","muffin_etc","rice","other_grains","noodles","pasta",
	"cakes_etc","sweet_pastries","puddings","sweet_biscuits","cream_biscuits","savoury_biscuits","savoury_combinations","sweet_combinations",
	"snack_noodles","fruit_bars","snack_bars","muesli_bars",
	"mince","beef_lamb_wovege","beef_lamb_wvege","plain_meat_wovege","plain_meat_wvege",
	"chicken_wovege","chicken_wvege","crumbed_chicken","plain_chicken_wovege","plain_chicken_wvege",
	"pork_wovege","pork_wvege","plain_pork_wovege","plain_pork_wvege","liver",
	"crumbed_fish","fresh_fish","canned_fish","other_seafood",
	"creamy_soup","clear_soup",
	"tacos_etc","sausages_etc","hamburgers","pizza","pie_etc","hot_dog","savoury_pastries","hash_browns_etc",
	"twisties_etc","potato_crisps_etc",
	"creamy_iceblock","water_iceblock",
	"chocolate","lollies",
	"low_fat_dressing","mayonnaise",
	"nuts","jam_etc","peanut_butter_etc","vegemite_etc","tomato_sauce_etc","devon_etc","bacon_etc","eggs",
	"jelly","takeaway_fries","home_fires",
	"potato","pumpkin","sweet_potato","cauliflower","green_beans","spinach","cabbage_etc",
	"peas","broccoli","carrots","zucchini_etc","capsicum","corn_etc","mushrooms","tomatoes",
	"lettuce","celery_etc","avocado","onion_etc",
	"soybeans_etc","baked_beans","other_beans_etc",
	"canned_fruit","fruit_salad","dried_fruit",
	"apple_etc","orange_etc","banana","peach_etc","mango_etc","pineapple","grapes_etc","melon")

micro_col <- c("fibre","thiamin","riboflavin","niacinequiv","vitc","folate","vita","retinol","betacarotine","sodium","potassium","magnesium","calcium","phosphorus","iron","zinc")
macro_col <- c("protein","fats","satfats","polyfats","monofats","cholesterol","carbohydrate","sugars")

aes$grapes_etc[which(aes$grapes_etc == "31")] <- "Never"

```

```{r, eval = FALSE}

# Get the factor level orders
aes_food_order <- sapply(aes[,food_col], order_food_levels)

# Order the values within the dataframe
for (i in 1:length(food_col)) {
	aes[,food_col[i]] <- ordered(aes[,food_col[i]], levels = aes_food_order[[i]])	
}

```

# Basic demographics

```{r, message=FALSE}

# Cases vs sibling vs control
print("ASD vs Sibling vs Unrelated")
table(aes$participant_type)

# Who completed survey
print("Who completed survey")
table(aes$who_completedsurvey)

# Age
ggplot(aes, aes(x = age, colour = participant_type, fill = participant_type)) + 
    geom_histogram(alpha = 0.1) +
    labs(legend.title = "Participant") +
    theme(legend.position = "bottom")

# Sex
tmp <- unique(aes$participant_type)
tmp.ls <- list()
for (i in 1:length(unique(tmp))) {
  tmp.ls[[i]] <- aes %>% filter(participant_type==tmp[i]) %>% dplyr::select("sex") %>% table()
}
foo <- do.call(rbind.data.frame, tmp.ls)
sex <- data.frame(Count=c(foo[,1], foo[,2]), 
                  Sex=c(rep("Female", length(tmp)), rep("Male", length(tmp))),
                  Group=rep(tmp,2))

# Plot
ggplot(sex, aes(x = Sex, y = Count, fill = Group)) + 
  geom_bar(stat = "identity", position = position_dodge())

```

# Diversity in diet

```{r}

diet_diversity <- data.frame(IID = aes$id, 
	participant_group = as.factor(aes$participant_group), 
	participant_type = as.factor(aes$participant_type), 
	age = aes$age, sex = aes$sex,
	shannon_micro = diversity(sapply(aes[,micro_col], as.numeric), "shannon"),
	shannon_macro = diversity(sapply(aes[,macro_col], as.numeric), "shannon"),
	shannon_food = diversity(sapply(aes[,food_col], function(x) as.numeric(as.factor(x))), "shannon"))

```

## Plot

```{r}

# Plot
diet_diversity.long <- melt(diet_diversity, 
	id.vars = c("IID", "participant_group", "participant_type", "age", "sex"),
	measure.vars = c("shannon_micro", "shannon_macro", "shannon_food"),
	variable.name = "diet_variable",
	value.name = "shannon")

ggplot(diet_diversity.long, 
	aes(x = participant_type, y = shannon, colour = participant_type)) + 
	geom_boxplot() +
	facet_grid(cols = vars(diet_variable))

# Focus on foods
ggplot(diet_diversity.long %>% filter(diet_variable == "shannon_food"), 
	aes(x = participant_type, y = shannon, colour = participant_type)) + 
	geom_boxplot() +
	facet_grid(cols = vars(diet_variable))

ggplot(diet_diversity.long %>% filter(diet_variable == "shannon_food"), 
	aes(x = participant_group, y = shannon, colour = participant_group)) + 
	geom_boxplot() +
	facet_grid(cols = vars(diet_variable))

```

## Statistical tests

- ANOVA (unadjusted + adjusted)
- Levene's test for difference in variance
- DO see differences! In both means and variance

```{r}

# ANOVA
# 1-way ANOVA

# Unadjusted for covariates
food.aov <- aov(shannon_food ~ participant_type, data = diet_diversity)
summary(food.aov)
food.aov <- aov(shannon_food ~ participant_group, data = diet_diversity)
summary(food.aov)

# Adjusted
food_diversity <- diet_diversity %>% filter(!is.na(shannon_food))
food_diversity$shannon_food_resid <- residuals(lm(shannon_food ~ age + as.factor(sex), data = food_diversity))
food.aov <- aov(shannon_food_resid ~ participant_type, data = food_diversity)
summary(food.aov)
food_diversity <- diet_diversity %>% filter(!is.na(shannon_food))
food_diversity$shannon_food_resid <- residuals(lm(shannon_food ~ age + as.factor(sex), data = food_diversity))
food.aov <- aov(shannon_food_resid ~ participant_group, data = food_diversity)
summary(food.aov)

# Levene's test for difference in variance
leveneTest(shannon_food_resid ~ participant_group, data = food_diversity)

```

# Metric 1: Macronutrients and general diet

- eg. carbohydrates, protein, fats, water intake

## Graph

```{r}
# Subset macronutrients table
aes_macro <- aes %>% 
  dplyr::select(id,participant_group,age,sex,weight,height,energy,protein,fats,satfats,polyfats,monofats,cholesterol,carbohydrate,sugars,water)

aes_macro[,5:ncol(aes_macro)] <- sapply(aes_macro[,5:ncol(aes_macro)], as.numeric)

# Show the table
datatable(aes_macro %>% head(10) %>% select(-"id"), rownames = FALSE, filter="top", options = list(pageLength = 10, scrollX=T))

# Correlation plot
# library(corrplot)
correlations <- cor(aes_macro[,c(5,6,8:ncol(aes_macro))]) 
corrplot(correlations, order = "hclust", method = "circle")

# Distributions of continuous variables
# library(caret)
y <- as.factor(aes_macro$participant_group)
x <- aes_macro[,c(5,6,8:ncol(aes_macro))]
scales <- list(x=list(relation="free"), y=list(relation="free"))
featurePlot(x=x, y=y, plot="density", scales=scales, auto.key=list(columns=3))

```

## PCA plot https://www.datacamp.com/community/tutorials/pca-analysis-r

```{r}

aes_macro.pca <- prcomp(aes_macro[,c(8:ncol(aes_macro))], center = T, scale = T)
aes_macro.ind <- aes_macro$participant_group
ggbiplot::ggbiplot(aes_macro.pca, groups=aes_macro.ind)

```

# Metric 2: Micronutrients

- eg. vitamins and minerals

## Graph
```{r}
# Subset micronutrients table
aes_micro <- aes %>% 
  dplyr::select(id,participant_group,age,sex,weight,height,fibre,thiamin,riboflavin,niacin,niacinequiv,vitc,folate,vita,retinol,betacarotine,sodium,potassium,magnesium,calcium,phosphorus,iron,zinc)

aes_micro[,5:ncol(aes_micro)] <- sapply(aes_micro[,5:ncol(aes_micro)], as.numeric)

# Show the table
datatable(aes_micro %>% head(10) %>% select(-"id"), rownames = FALSE, filter="top", options = list(pageLength = 10, scrollX=T) )

# Correlation plot
# library(corrplot)
correlations <- cor(aes_micro[,c(5,6,8:ncol(aes_micro))]) 
corrplot(correlations, order = "hclust", method = "circle")

# Distributions of continuous variables
# library(caret)
y <- as.factor(aes_micro$participant_group)
x <- aes_micro[,c(5,6,8:ncol(aes_micro))]
scales <- list(x=list(relation="free"), y=list(relation="free"))
featurePlot(x=x, y=y, plot="density", scales=scales, auto.key=list(columns=3))
featurePlot(x=x, y=y, plot="strip", scales=scales, auto.key=list(columns=3))

```

## PCA plot
```{r}

aes_micro.pca <- prcomp(aes_micro[,c(8:ncol(aes_micro))], center = T, scale = T)
aes_micro.ind <- aes_micro$participant_group
ggbiplot::ggbiplot(aes_micro.pca, groups=aes_micro.ind)

```

# Metric 3: Proportion of energy 
## By macronutrient
ie. what proportion of energy is contributed to by each macronutrient

```{r}
# Display data
aes_pe_macro <- aes[,c("id", "participant_group", "age", "peprotein", "pecarbohydrate", "pefats", "pesatfats", "pepolyfats", "pemonofats")]

aes_pe_macro[,3:ncol(aes_pe_macro)] <- sapply(aes_pe_macro[,3:ncol(aes_pe_macro)], as.numeric)

datatable(aes_pe_macro %>% head(10) %>% select(-"id"), rownames = FALSE, filter="top", options = list(pageLength = 10, scrollX=T) )
```

### Graph
```{r}
# Box and whisker plot
aes_pe_macro.long <- melt(aes_pe_macro, id.vars=c("participant_group"), measure.vars=c("peprotein", "pecarbohydrate", "pefats", "pesatfats", "pepolyfats", "pemonofats"), variable.name="macronutrient")
colnames(aes_pe_macro.long) <- c("group", "macronutrient", "prop_energy")
aes_pe_macro.gg <- ggplot(aes_pe_macro.long, aes(x=macronutrient, y=prop_energy, colour=group)) +
  geom_boxplot()
aes_pe_macro.gg
```

## By core (ie. healthy) vs. non-core (unhealthy) food
ie. what proportion of energy is contributed to by healthy foods (aka "core" foods)

### Graph

```{r}
aes_pe_core <- aes %>% dplyr::select(id, participant_group, age, pecore, penoncore)

aes_pe_core[,3:ncol(aes_pe_core)] <- sapply(aes_pe_core[,3:ncol(aes_pe_core)], as.numeric)

# Box and whisker plot
aes_pe_core.long <- melt(aes_pe_core, id.vars=c("participant_group"), measure.vars=c("pecore", "penoncore"), variable.name="core_noncore")
colnames(aes_pe_core.long) <- c("group", "core_noncore", "prop_energy")
aes_pe_core.gg <- ggplot(aes_pe_core.long, aes(x=core_noncore, y=prop_energy, colour=group)) +
  geom_boxplot()
aes_pe_core.gg
```

## PCA plot

```{r}

aes_pe_core.pca <- prcomp(aes_pe_core[,c(3:5)], center = T, scale = T)
aes_pe_core.ind <- aes_pe_core$participant_group
ggbiplot::ggbiplot(aes_pe_core.pca, groups=aes_pe_core.ind)

```

## By food groups

- Comparing groups with respect to what proportion of energy they gain from each food group

Can break down into:

- Core foods: healthy eg. veg, fruit, meat, protein alternatives, grains, dairy
- Non-core: unhealthy eg. sweet drink, packed snack, confectionery, baked products, takeaway, condiments, fatty meats

### Graph

```{r}
aes_pe_spec <- aes %>% dplyr::select(participant_group, age, peveg, pefruit, pemeat, pealt, pegrains, pedairy, pesweet_drink, pepack_snack, peconfect, pebaked_products, petakeaway, pecondiments, pefatty_meats)

aes_pe_spec[,3:ncol(aes_pe_spec)] <- sapply(aes_pe_spec[,3:ncol(aes_pe_spec)], as.numeric)

# Show the table
datatable(aes_pe_spec %>% head(10), rownames = FALSE, filter="top", options = list(pageLength = 10, scrollX=T))

# Correlation plot
# library(corrplot)
correlations <- cor(aes_pe_spec[,c(3:ncol(aes_pe_spec))]) 
corrplot(correlations, order = "hclust", method = "circle")

# Distributions of continuous variables
# library(caret)
y <- as.factor(aes_pe_spec$participant_group)
x <- aes_pe_spec[,c(4:ncol(aes_pe_spec))]
scales <- list(x=list(relation="free"), y=list(relation="free"))
featurePlot(x=x, y=y, plot="density", scales=scales, auto.key=list(columns=3))

# Box and whisker plot
#aes_pe_spec2 <- aes_pe_spec %>% select(-age) %>% select(-id)
aes_pe_spec.long <- melt(aes_pe_spec, id.vars=c("participant_group"), variable.name="specific_food")
colnames(aes_pe_spec.long) <- c("group", "specific_food", "prop_energy")
aes_pe_spec.gg <- ggplot(aes_pe_spec.long, aes(x=specific_food, y=prop_energy, colour=group)) +
  geom_boxplot() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
aes_pe_spec.gg

# Remove the ID column
aes_pe_spec <- aes_pe_spec %>% dplyr::select(-age)
```

## PCA plot

```{r}
aes_pe_spec.pca <- prcomp(aes_pe_spec[,c(3:ncol(aes_pe_spec))], center = T, scale = T)
aes_pe_spec.ind <- aes_pe_spec$participant_group
ggbiplot::ggbiplot(aes_pe_spec.pca, groups=aes_pe_spec.ind)
```

# Metric 4: Australian Recommended Food Score (ARFS)

```{r}
# Display data
aes_arfs <- aes %>% 
  dplyr::select(participant_group, arfs, arfs_veg,arfs_fruit, arfs_meat, arfs_alt, arfs_grains, arfs_dairy, arfs_extras)

aes_arfs[,2:ncol(aes_arfs)] <- sapply(aes_arfs[,2:ncol(aes_arfs)], as.numeric)

datatable(aes_arfs %>% head(10), rownames = FALSE, filter="top", options = list(pageLength = 10, scrollX=T) )
```

## Graph

```{r}
# Box and whisker plot
aes_arfs.long <- melt(aes_arfs, id.vars=c("participant_group"), variable.name="arfs_group")
colnames(aes_arfs.long) <- c("group", "arfs_group", "score")
aes_arfs.gg <- ggplot(aes_arfs.long, aes(x=arfs_group, y=score, colour=group)) +
  geom_boxplot() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
aes_arfs.gg
```

# PC generation: CLR, % energy data, no covariate regression + add Bristol Stool Chart

- As above, but apply clr transform to %energy data as it's proportional

## Centred-log ratio transformation

```{r}

library(mixOmics)

aes_1193652_pe <- aes_1193652 %>% dplyr::select(id, participant_group, month, age, sex, bmi, peveg, pefruit, pemeat, pealt, pegrains, pedairy, pesweet_drink, pepack_snack, peconfect, pebaked_products, petakeaway, pecondiments, pefatty_meats)

aes_1193652_pe[,7:ncol(aes_1193652_pe)] <- sapply(aes_1193652_pe[,7:ncol(aes_1193652_pe)], as.numeric)

aes_1193652_pe.clr.tmp <- logratio.transfo(as.matrix(aes_1193652_pe[,7:ncol(aes_1193652_pe)]), logratio = "CLR", offset = 0.001)
class(aes_1193652_pe.clr.tmp) <- "matrix"
#aes_1193652_pe.clr.tmp <- clr(as.matrix(aes_1193652_pe[,7:ncol(aes_1193652_pe)]))
aes_1193652_pe.clr <- data.frame(aes_1193652_pe[,1:5], aes_1193652_pe.clr.tmp)

# Distributions of continuous variables
y <- as.factor(aes_1193652_pe.clr$participant_group)
x <- aes_1193652_pe.clr[,c(6:ncol(aes_1193652_pe.clr))]
scales <- list(x=list(relation="free"), y=list(relation="free"))
featurePlot(x=x, y=y, plot="density", scales=scales, auto.key=list(columns=3))

```

## Generate PCs

```{r}

aes_1193652_pe.clr.pca <- prcomp(aes_1193652_pe.clr[,c(6:ncol(aes_1193652_pe.clr))], center = T, scale = T)
aes_1193652_pe.clr.grp <- aes_1193652_pe.clr$participant_group
aes_1193652_pe.clr.ind <- aes_1193652_pe.clr$id
aes_1193652_pe.clr.age <- as.factor(round(aes_1193652_pe.clr$age, 0))

# Generate plots
ggbiplot::ggbiplot(aes_1193652_pe.clr.pca, groups=aes_1193652_pe.clr.grp, labels=aes_1193652_pe.clr.ind, ellipse = TRUE)

```

## Check PC interpretability

```{r}

# PC loadings per questionnaire
aes_1193652_pe.clr.lod <- aes_1193652_pe.clr.pca$rotation
datatable(aes_1193652_pe.clr.lod) %>% formatRound("PC1", 3) %>% formatRound("PC2", 3) %>% formatRound("PC3", 3) %>% formatRound("PC4", 3) %>% formatRound("PC5", 3) %>% formatRound("PC6", 3) %>% formatRound("PC7", 3) %>% formatRound("PC8", 3) %>% formatRound("PC9", 3) %>% formatRound("PC10", 3) %>% formatRound("PC11", 3) %>% formatRound("PC12", 3) %>% formatRound("PC13", 3) 

aes_1193652_pe.clr.lod.long <- melt(aes_1193652_pe.clr.lod)
colnames(aes_1193652_pe.clr.lod.long) <- c("Food_group", "PC", "Loading")
ggplot(data = aes_1193652_pe.clr.lod.long, aes(x = PC, y = Loading, fill = Food_group)) +
  geom_bar(stat="identity", position=position_dodge())
aes_1193652_pe.clr.lod.long1to5 <- aes_1193652_pe.clr.lod.long %>% filter(PC %in% c("PC1", "PC2", "PC3", "PC4", "PC5"))
ggplot(data = aes_1193652_pe.clr.lod.long1to5, aes(x = PC, y = Loading, fill = Food_group)) + 
  geom_bar(stat="identity", position=position_dodge())

```

## Check variance explained per PC

```{r}

fviz_eig(aes_1193652_pe.clr.pca)

print(summary(aes_1193652_pe.clr.pca))

```

### Compare PCs by group

```{r}

aes_1193652_pe.clr.pcs <- aes_1193652_pe.clr.pca$x
aes_1193652_pe.clr.pcs <- data.frame(id = aes_1193652_pe.clr.ind, participant_group = aes_1193652_pe.clr$participant_group, aes_1193652_pe.clr.pcs)

aes_1193652_pe.clr.pcs %>% 
  group_by(participant_group) %>%
  dplyr::summarise(mean_PC1 = round(mean(PC1),2), sd_PC1 = round(sd(PC1),2),
                   mean_PC2 = round(mean(PC2),2), sd_PC2 = round(sd(PC2),2),
                   mean_PC3 = round(mean(PC3),2), sd_PC3 = round(sd(PC3),2))

# Check for differences between UNR and UNR_QTAB groups
aes_1193652_pe.clr.pcs2 <- aes_1193652_pe.clr.pcs
aes_1193652_pe.clr.pcs2$participant_group <- as.character(aes_1193652_pe.clr.pcs2$participant_group)
aes_1193652_pe.clr.pcs2$id <- as.character(aes_1193652_pe.clr.pcs2$id)
aes_1193652_pe.clr.pcs2[which(nchar(aes_1193652_pe.clr.pcs2$id) == 6),"participant_group"] <- "UNR_QTAB"

aes_1193652_pe.clr.pcs2 %>% 
  group_by(participant_group) %>%
  dplyr::summarise(mean_PC1 = round(mean(PC1),2), sd_PC1 = round(sd(PC1),2),
                   mean_PC2 = round(mean(PC2),2), sd_PC2 = round(sd(PC2),2),
                   mean_PC3 = round(mean(PC3),2), sd_PC3 = round(sd(PC3),2))

```

<br><br>
